src/Compilers/Core/Portable/Symbols/Attributes/AttributeDescription.cs

```diff
@@ -28,6 +28,12 @@ public AttributeDescription(string @namespace, string name, byte[][] signatures,
             this.MatchIgnoringCase = matchIgnoringCase;
         }
 
+        public bool Equals(AttributeDescription other)
+            => this.Namespace == other.Namespace &&
+                this.Name == other.Name &&
+                ReferenceEquals(this.Signatures, other.Signatures) &&
+                this.MatchIgnoringCase == other.MatchIgnoringCase;
```

Comment:
Consider moving this to `NullableWalker.MethodHasAttribute` since the method is only used for that (temporary?) scenario.